### PR TITLE
Support redirect port for azd login

### PR DESCRIPTION
**Pull request by vhvb1989**
This PR adds a new parameter `redirect-port` for `azd login` which can be used to define the port to use after doing an interactive web browser authentication.

The default behavior (when this new arg is not set, or is set to 0) is to let `azd` to automatically select an available port.
 
But, in some cases, customers might want to define the port to be used for convenience. For example:
- When using codespaces, customers can create a `port redirect` before calling `azd login`.
- When using remote-ssh connection, similarly, customers don't need to wait for the url redirect from `azd login` to discover the port that azd is using.
- When using a `dev-tunnel`, each new `port redirect` requires approving the connections thru the tunnel (the first time it is used). This approval ends up as an invalid response to the `MSAL client` on azd waiting for the token response and `azd` fails to complete login.  So, if the port is known in advance, customer can approve the connection before calling `azd login` and use the already approved tunnel to complete the login

### CODE REVIEW

File: cli/azd/cmd/login.go [hunk 1]
```diff
@@ -95,6 +96,7 @@ func (lf *loginFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandO
 		"",
 		"The provider to use to acquire a federated token to authenticate with.")
 	local.StringVar(&lf.tenantID, "tenant-id", "", "The tenant id for the service principal to authenticate with.")
+	local.IntVar(&lf.redirectPort, "redirect-port", 0, "Choose the port to use when doing interactive login.")
```
Comment (weikanglim) [inline cli/azd/cmd/login.go:99]: I'd vote for preciseness given that this is an advanced user feature 😄 

```suggestion
	local.IntVar(&lf.redirectPort, "redirect-port", 0, "Choose the port to be used as part of the redirect URI during interactive login.")
```

**Review by weikanglim — APPROVED** [review]
LGTM

**Review by ellismg — APPROVED** [review]
LGTM - Nice improvement!

**Pull request merged**